org.lflang/src/org/lflang/generator/c/CGenerator.java

```diff
@@ -563,6 +564,22 @@ public void doGenerate(Resource resource, LFGeneratorContext context) {
                     dockerGenerator.fromData(lfModuleName, federate.name, fileConfig));
             }
 
+            System.out.println("Checking main instance in CGenerator.java.");
+            System.out.println(this.main);
+            System.out.println(main);
+
+            // Generate schedule.h if QS scheduler is used.
```

Comment:
Sorry to repeat this, but this should also be in [`pickScheduler`](https://github.com/lf-lang/lingua-franca/blob/d561add3e18a2e65b8648f240a4c7609819eb653/org.lflang/src/org/lflang/generator/c/CGenerator.java#L864) I think.

Reply:
Make sense. Will revise the code accordingly. Thanks.